openedx/core/djangoapps/programs/docs/decisions/0001-sync-certificate-available-dates.rst

```diff
@@ -0,0 +1,46 @@
+Sync certificate_available_date and visible_date for certificates
+=================================================================
+
+Status
+------
+
+Review
+
+Context
+-------
+
+In the Credentials service, certificate_available_date is called visible_date.
+Whenever a generatedCertificate is updated or saved the first time, it sets
+the visible date to certificate available date (available in studio settings).
+
+There is a limitation in our code, if course team decides to change the
+certificate_available_date it is not reflected in the credentials service
+for all the certificates that were awarded before that change and have been
+pushed to credentials service. The certificates' dates need to be in sync on
+both places.
+
+This is causing problem because a learner is able to see the certificate on
+their dashboard but the change is not reflected on their learner record in
+credential service.
+
+Decision
+--------
+
+Whenever a CourseUpdate is made with change to the certificate_available_date,
+we need to update certificate visible_date on the credential service.
+
+To achieve this, we need a signal to listen for any course update that contains
```

Comment:
`To achieve this, a signal is created to listen to any course update signal that contains...`